pkgs/development/tools/build-managers/bazel/bazel_7/default.nix

```diff
@@ -117,7 +100,137 @@ let
       unzip
       which
       zip
+      runJdk
+      makeWrapper
+    ];
+
+  # Bootstrap an existing Bazel so we can vendor deps with vendor mode
+  bazel_bootstrap = stdenv.mkDerivation rec {
+    name = "bazel_bootstrap";
+
+    src = if stdenv.hostPlatform.system == "x86_64-linux" then fetchurl {
+      url = "https://github.com/bazelbuild/bazel/releases/download/${version}/bazel_nojdk-${version}-linux-x86_64";
+      hash = "sha256-05fHtz47OilpOVYawB17VRVEDpycfYTIHBmwYCOyPjI=";
+    } else if stdenv.hostPlatform.system == "aarch64-linux" then fetchurl {
+      url = "https://github.com/bazelbuild/bazel/releases/download/${version}/bazel_nojdk-${version}-linux-arm64";
+      hash = "sha256-olrlIia/oXWleXp12E+LGXv+F1m4/S4jj/t7p2/xGdM=";
+    } else if stdenv.hostPlatform.system == "x86_64-darwin" then fetchurl {
+      url = "https://github.com/bazelbuild/bazel/releases/download/${version}/bazel_nojdk-${version}-darwin-x86_64";
+      hash = "sha256-e5RRZTrhzoTMk2NRRl9m2K5+h1/+5mJNW2vF8h0uftc=";
+    } else fetchurl { # stdenv.hostPlatform.system == "aarch64-darwin"
+        url = "https://github.com/bazelbuild/bazel/releases/download/${version}/bazel_nojdk-${version}-darwin-arm64";
+        hash = "sha256-ciNBdqm+cIrNR1Wu2wt+wGGb6m9i9aE8LhRqL4BJ0Y4=";
+    };
+
+    nativeBuildInputs = defaultShellUtils;
+    buildInputs = [
+      stdenv.cc.cc
+      autoPatchelfHook
+    ];
+
+    dontUnpack = true;
+    dontPatch = true;
+    dontBuild = true;
+    dontStrip = true;
+    installPhase = ''
+      runHook preInstall
+
+      mkdir -p $out/bin
+      install -Dm755 $src $out/bin/bazel
+
+      runHook postInstall
+    '';
+
+    postFixup = ''
+      wrapProgram $out/bin/bazel \
+        --prefix PATH : ${lib.makeBinPath nativeBuildInputs}
+    '';
+
+    meta.sourceProvenance = with lib.sourceTypes; [ binaryNativeCode ];
+  };
+
+  # The bootstrapped Bazel did not work on its own for vendoring, but a wrapped FHS did
```

Comment:
Can we add any detail on why that is here? I assume it might be due to Bazel making lookups at fixed paths only?

Reply:
Unfortunately, debugging this issue was a bit opaque. Bazel has a plethora of configuration settings, and it's hard to determine which are necessary in situations like this. I knew about the FHS trick from the (now sadly private) sourcegraph repo, so I tried it out and found it worked. We may be able to simplify this a bit, or I can remove the comment since it is more editorial than descriptive.